lib-multisrc/libgroup/src/eu/kanade/tachiyomi/multisrc/libgroup/LibGroup.kt

```diff
@@ -716,38 +478,51 @@ abstract class LibGroup(
         arrayOf("Да", "Все"),
     )
 
+    // Utils
+    private inline fun <reified T> String.parseAs(): T = json.decodeFromString(this)
+
+    private inline fun <reified T> Response.parseAs(): T = body.string().parseAs()
+
+    private fun urlChangedError(sourceName: String): String =
+        "URL серии изменился. Перенесите с $sourceName " +
+            "на $sourceName, чтобы обновить URL-адрес."
+
+    private val scope = CoroutineScope(Dispatchers.IO)
+    private fun launchIO(block: () -> Unit) = scope.launch { block() }
+
     companion object {
         const val PREFIX_SLUG_SEARCH = "slug:"
         private const val SERVER_PREF = "MangaLibImageServer"
 
         private const val SORTING_PREF = "MangaLibSorting"
-        private const val SORTING_PREF_Title = "Способ выбора переводчиков"
+        private const val SORTING_PREF_TITLE = "Способ выбора переводчиков"
 
-        private const val isScan_USER = "ScanlatorUsername"
-        private const val isScan_USER_Title = "Альтернативный переводчик"
+        private const val IS_SCAN_USER = "ScanlatorUsername"
+        private const val IS_SCAN_USER_TITLE = "Альтернативный переводчик"
 
         private const val TRANSLATORS_TITLE = "Чёрный список переводчиков\n(для красоты через «/» или с новой строки)"
         private const val TRANSLATORS_DEFAULT = ""
 
         private const val LANGUAGE_PREF = "MangaLibTitleLanguage"
-        private const val LANGUAGE_PREF_Title = "Выбор языка на обложке"
+        private const val LANGUAGE_PREF_TITLE = "Выбор языка на обложке"
 
-        private val simpleDateFormat by lazy { SimpleDateFormat("yyyy-MM-dd", Locale.US) }
+        val simpleDateFormat by lazy { SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'", Locale.US) }
     }
 
-    private var isServer: String? = preferences.getString(SERVER_PREF, "fourth")
+    private var isServer: String? = preferences.getString(SERVER_PREF, "main")
     private var isEng: String? = preferences.getString(LANGUAGE_PREF, "eng")
     private var groupTranslates: String = preferences.getString(TRANSLATORS_TITLE, TRANSLATORS_DEFAULT)!!
+    private var isScanUser: Boolean? = preferences.getBoolean(IS_SCAN_USER, false)
```

Comment:
this can cause `preferences` to be initialized before the id if it is overridden in the source class, change it to a function instead